### PR TITLE
Add edit link to workshops

### PR DESCRIPTION
**Pull request by jessicard**
Position is fixed so it scrolls with you.

Screen Shot 2022-04-14 at 3 47 26 PM

### CODE REVIEW

**Comment by jessicard** [comment]
Not super pumped about how it looks yet though. @exu3 do you have any ideas/advice for how this should look?

**Comment by jessicard** [comment]
One other question: right now it goes to https://github.com/hackclub/hackclub/tree/main/workshops/gamelab. Should it actually go to https://github.com/hackclub/hackclub/edit/main/workshops/gamelab/README.md?

**Comment by jessicard** [comment]
@maxwofford agreed about going directly to the edit page, done in [5b27be2](https://github.com/hackclub/workshops/pull/99/commits/5b27be24a2d57665c21c5dfdc329e9642e9f29fe)

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:


| [**/\[slug\]**](https://workshops-b456yms04.hackclub.dev/[slug]) |
|:-:|
|<a href="https://workshops-b456yms04.hackclub.dev/[slug]">Screenshot of /[slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/hpfMJuyKKTBILnpJ4vaM2toxg00Pj3U_FfdwanlYQhkapD0mSsgpCs3SgPem2UosSHJobRK6lX1EmND8p7riTg.png">(view full size)</a> |







Commit 5b27be24a2d57665c21c5dfdc329e9642e9f29fe (https://workshops-b456yms04.hackclub.dev).

**Comment by exu3** [comment]
> Not super pumped about how it looks yet though. @exu3 do you have any ideas/advice for how this should look?

Maybe a `Button` reading something like "Edit this page on GitHub" with the GitHub icon?

**Comment by cjdenio** [comment]
Here's an idea:

image

**Comment by jessicard** [comment]
@exu3 Great idea!

@cjdenio That's perfect!! pls feel free to push to the branch 🙏

**Comment by sampoder** [comment]
This PR looks sick!! Some text on mobile is a bit difficult to read though: 



Maybe we could remove the transparent background or make it just the icon on mobile?

**Comment by jessicard** [comment]
@sampoder Ooooh good catch!! I added a bg color in [72e9445](https://github.com/hackclub/workshops/pull/99/commits/72e94458d3b8e69d7d3362ab71664248d9c46960).

I like the idea of potentially shortening it to just the icon. Is there another place in this app where we check for mobile/screen size you could point me to?

Screen Shot 2022-04-15 at 11 22 41 AM

**Comment by sampoder** [comment]
On the train rn so I don't have a super easy way of Browsing the code but check out https://theme-ui.com/sx-prop/ (the responsive values section)

**Comment by sampoder** [comment]
Also for the background colour, instead of "white" let's use "background" which will change based on dark/light mode

**Comment by sampoder** [comment]
for responsive thing you'll want the text to have something like: `display: ["none", "inline"]` replace inline with what ever it currently is and it will hide on mobile but show on tablets and anything larger

File: pages/[slug]/index.js [hunk 1]
```diff
@@ -44,6 +47,28 @@ const Page = ({ slug, data, html, locales }) => {
...
+        <Button
+          as="a"
+          variant="outline"
+          href={`https://github.com/hackclub/hackclub/edit/main/workshops/${slug}/README.md`}
+          sx={{
+            backgroundColor: 'background',
+          }}
+        >
+          <Edit3 />
+          Edit this page!
```
Comment (sampoder) [inline pages/[slug]/index.js:68]: probs wrap this in a `<Text>` component with `sx={{//the stuff I sent early}}`

Comment (jessicard) [reply pages/[slug]/index.js:68]: @sampoder Perfect, that's working great, thank you! Your instructions were very helpful 🙏 

Last thing I'm messing with is that there's a margin on the edit SVG that i'm trying to get rid of if the text isn't there, but it doesn't seem to be applying the styles from `sx`. I'm new to `theme-ui` so I'm clunking through 😅, thanks for your help!

Comment (sampoder) [reply pages/[slug]/index.js:68]: Ahh I think that's a react feather icon which is separate... I'll look into it

Comment (sampoder) [reply pages/[slug]/index.js:68]: Could you send me a screenshot of the margin?

Comment (jessicard) [reply pages/[slug]/index.js:68]: Here you go! Specifically it's the `margin-right` we want to set to `0`.

Screen Shot 2022-04-15 at 11 45 55 AM

Comment (cjdenio) [reply pages/[slug]/index.js:68]: That's set by the theme: https://github.com/hackclub/theme/blob/95a382891132241dc3a517d091554da1e482e339/packages/theme/src/index.ts#L215

Comment (sampoder) [reply pages/[slug]/index.js:68]: oh yeah good catch!

**Comment by jessicard** [comment]
Thanks y'all! Screen Shot 2022-04-18 at 10 38 57 AM
Screen Shot 2022-04-18 at 10 38 52 AM

**Review by cjdenio — APPROVED** [review]
Looks great!

**Comment by sampoder** [comment]
YAY!

**Pull request merged**